receiver/awscontainerinsightreceiver/internal/stores/podstore.go

```diff
@@ -306,6 +311,12 @@ func (p *PodStore) refreshInternal(now time.Time, podList []corev1.Pod) {
 			cpuRequest += tmpCPUReq
 			tmpMemReq, _ := getResourceSettingForPod(&pod, p.nodeInfo.getMemCapacity(), memoryKey, getRequestForContainer)
 			memRequest += tmpMemReq
+			if tmpGpuReq, ok := getResourceSettingForPod(&pod, 0, gpuKey, getRequestForContainer); ok {
+				gpuRequest += tmpGpuReq
+			}
+			if tmpGpuLimit, ok := getResourceSettingForPod(&pod, 0, gpuKey, getLimitForContainer); ok && pod.Status.Phase == corev1.PodRunning {
```

Comment:
would it make sense to combine `request` and `limit` if blocks into single `limit` then handle `gpuRequest` in the consolidated if block since they should be equal anyways? `PodRunning` check can be nested if block inside.

Reply:
Updated